File: DSA/RATNA_KHANNA_590013141/DAY_19_QUESTION_2.c
#include <stdio.h>

int findPeak(int arr[], int low, int high, int n) {
    int mid = low + (high - low) / 2;

    if ((mid == 0 || arr[mid - 1] <= arr[mid]) && (mid == n - 1 || arr[mid + 1] <= arr[mid]))
        return mid;

    if (mid > 0 && arr[mid - 1] > arr[mid])
        return findPeak(arr, low, mid - 1, n);

    return findPeak(arr, mid + 1, high, n);
}

int main() {
    int n;

    printf("Enter the size of the array: ");
    scanf("%d", &n);

    int arr[n];

    printf("Enter the elements of the array:\n");
    for (int i = 0; i < n; i++) {
        scanf("%d", &arr[i]);
    }

    int peakIndex = findPeak(arr, 0, n - 1, n);
    printf("Peak element found at index: %d\n", peakIndex);

    return 0;
}

File: DSA/pranavdixit_590014418/Day_19_Question_2.py
def find_peak(arr):
    low, high = 0, len(arr) - 1
    while low < high:
        mid = (low + high) // 2
        if arr[mid] < arr[mid + 1]:
            low = mid + 1
        else:
            high = mid
    return low

print(find_peak([1, 3, 20, 4, 1, 0]))
print(find_peak([1, 2, 1, 3, 5, 6, 4]))
print(find_peak([10, 20, 15, 2, 23, 90, 67]))
